modules/gdscript/gdscript_parser.cpp

```diff
@@ -458,9 +458,9 @@ GDScriptParser::Node *GDScriptParser::_parse_expression(Node *p_parent, bool p_s
 			if (!validating) {
 
 				//this can be too slow for just validating code
-				if (for_completion && ScriptCodeCompletionCache::get_singleton()) {
+				if (for_completion && ScriptCodeCompletionCache::get_singleton() && FileAccess::exists(path)) {
 					res = ScriptCodeCompletionCache::get_singleton()->get_cached_resource(path);
-				} else { // essential; see issue 15902
+				} else if ((!for_completion) || FileAccess::exists(path)) {
```

Comment:
That's a style matter, but the parentheses around `!for_completion` are not necessary.

Reply:
Ah ok, I'm used to guidelines that require me to be paranoid about anyone potentially misunderstanding the order in which things are evaluated.
Should I change it to conform the preferred style?

Reply:
If you can amend the commit, that would be more consistent with the style used in the rest of the code base yeah.